plugins/aws/appmesh/appmeshVGHealthChecks.js

```diff
@@ -0,0 +1,97 @@
+
+            if (!listMeshes) return rcb();
+
+            if (listMeshes.err || !listMeshes.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query for App Mesh meshes: ' + helpers.addError(listMeshes), region);
+                return rcb();
+            }
+
+            if (!listMeshes.data.length) {
+                helpers.addResult(results, 0, 'No App Mesh meshes found', region);
+                return rcb();
+            }
+
+            for (let mesh of listMeshes.data){
+                if (!mesh.arn || !mesh.meshName) continue;
+
+                let resource = mesh.arn;
+
+                var listVirtualGateways = helpers.addSource(cache, source,
+                    ['appmesh', 'listVirtualGateways', region, mesh.meshName]);
+
+                if (!listVirtualGateways || listVirtualGateways.err || !listVirtualGateways.data) {
+                    helpers.addResult(results, 3,
+                        'Unable to query for App Mesh virtual gateways: ' + helpers.addError(listVirtualGateways),
+                        region, resource);
+                    continue;
+                }
+
+                if (!listVirtualGateways.data.virtualGateways) {
+                    helpers.addResult(results, 0,
+                        'No App Mesh virtual gateways found', region, resource);
+                    continue;
+                }
+
+                for (let gateway of listVirtualGateways.data.virtualGateways) {
+                    if (!gateway.arn || !gateway.virtualGatewayName) continue;
+
+                    let resource = gateway.arn;
+
+                    var describeVirtualGateway = helpers.addSource(cache, source,
+                        ['appmesh', 'describeVirtualGateway', region, gateway.virtualGatewayName]);
+
+                    if (!describeVirtualGateway ||
+                        describeVirtualGateway.err ||
+                        !describeVirtualGateway.data) {
+                        helpers.addResult(results, 3,
+                            'Unable to describe App Mesh virtual gateway: ' + helpers.addError(describeVirtualGateway), region, resource);
+                        continue;
+                    }
+
+                    if (describeVirtualGateway.data.virtualGateway &&
+                        describeVirtualGateway.data.virtualGateway.spec &&
+                        describeVirtualGateway.data.virtualGateway.spec.listeners) {
+                        const listeners = describeVirtualGateway.data.virtualGateway.spec.listeners;
+
+                        const hasHealthCheckPolicies = listeners.every(listener => {
+                            return listener.healthCheck && listener.healthCheck.protocol &&
+                                listener.healthCheck.healthyThreshold && listener.healthCheck.intervalMillis &&
+                                listener.healthCheck.timeoutMillis && listener.healthCheck.unhealthyThreshold;
```

Comment:
```suggestion
                            return listener.healthCheck && Object.keys(listener.healthCheck).length;
```

maybe we should just check if healthCheck exists and not each field specifically